### PR TITLE
Update CONTRIBUTING

### PR DESCRIPTION
**Pull request by wemeetagain**
View the preview here: https://github.com/ChainSafe/lodestar/blob/cayman/style-guide/CONTRIBUTING.md

Suggestions welcome

### CODE REVIEW

**Review by ColinSchwarz — APPROVED** [review]
Everything looks good in terms proper English and clear organization. I can't speak much to the actual content since I've never wrote a contributor guideline for code but it all reads nicely and is succinct.

**Review by GregTheGreek — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,54 +1,51 @@
...
-
-# Code review process
-> The core team looks at Pull Requests on a regular basis. 
+> Thanks for your contribution to Lodestar. It's people like you that push the Ethereum ecosystem forward.
+
+# First-time Contributor?
+Unsure where to begin contributing to Lodestar? Here are some ideas!
+
+- See any typos? See any verbiage that should be changed or updated? Go for it! Github makes it easy to make contributions right from the browser.
+- Look through our [outstanding unassigned issues](https://github.com/ChainSafe/lodestar/issues?q=is%3Aopen+is%3Aissue+no%3Aassignee).
```
Comment (mpetrunic) [inline CONTRIBUTING.md:9]: Hint about good first issue label?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,54 +1,51 @@
...
+See [packages/](https://github.com/ChainSafe/lodestar/tree/master/packages) for a list of packages hosted in this repo.
+
+# Style Guide
+
+- Functions and variables should be [`camelCase`](https://en.wikipedia.org/wiki/Camel_case), classes should be [`PascalCase`](http://wiki.c2.com/?PascalCase), constants should be `UPPERCASE_WITH_UNDERSCORES`.
+- Use `"` instead of `'`
+- All functions should have types declared for all parameters and return value
+- All interfaces should be prefixed with a `I`
+  - eg: `IMyInterface`
+- You probably shouldn't be using Typescript's `any`
```
Comment (mpetrunic) [inline CONTRIBUTING.md:36]: I think we should enforce styleguide with eslint file and not here

**Review by mpetrunic — COMMENTED** [review]
damm, review was pending :( @wemeetagain

**Comment by mpetrunic** [comment]
> damm, review was pending :( @wemeetagain

I mean it's nothing major, we can fix it in subsequent PRs

**Comment by wemeetagain** [comment]
I'll make another one now :)

File: CONTRIBUTING.md (hunk 2 above)
Comment (wemeetagain) [reply CONTRIBUTING.md:36]: I figure this is an english description of our eslint rules + a few other rules.
Next PR to include eslint rule changes

Comment (mpetrunic) [reply CONTRIBUTING.md:36]: True, but it's unnecessary as CI won't allow them to merge and IDE will throw errors and annoy them anyways :joy:

Comment (mpetrunic) [reply CONTRIBUTING.md:36]: It doesn't really matter, I'm good if it stays haha

Comment (wemeetagain) [reply CONTRIBUTING.md:36]: I can move the big eslint rules under the "Make sure your code is properly linted" bullet point so its less clutter

**Pull request merged**